### PR TITLE
Format implementation section of DG

### PR DESCRIPTION
**Pull request by delishad21**
Currently, the implementation section of the DG is messy and the headers are not standardized in formatting. Lets update the implementation portion of the DG to make everything neater. 

This PR also adds additional information to the Effort section.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -112,6 +114,8 @@ With listeners, the update logic would be scattered across multiple UI component
 
 One of our main goals was to make our codebase easy to understand and maintain, and we felt that the centralized update method would be more in line with this goal despite the slight increase in coupling and inefficiency.
 
+<br>
```
Comment (Zer0Legion) [inline docs/DeveloperGuide.md:117]: Good use of the header to make use of whitespace

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -2331,34 +2363,101 @@ New fields such as recitation, studio, matriculation number, was added to person
...
-The implementation of the exam and exam score features was the most significant addition to our application, requiring adjustments to existing features and the 
-introduction of many new commands to handle and manage the addition of exams and exam features.
+The implementation of the exam and exam score features was the most significant addition to our application, requiring adjustments to existing features and the
+introduction of many new commands to handle and manage the addition of exams and exam features. This feature was the most
+complex and required the most effort to implement, as it involved the introduction of a new entity, `Exam`, and the management
+of scores for each person for each exam. This is further elaborated upon in the challenges section below.
+
+<br>
+
+### Challenges Faced
```
Comment (Zer0Legion) [inline docs/DeveloperGuide.md:2401]: Very detailed writeup!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -242,13 +258,12 @@ and optionally provide additional information such as their:
...
 
-#### Implementation Details
-
 ##### Parsing User Input
 The `AddCommandParser` class is responsible for parsing user input to extract the details of the person to be added. It uses the `ArgumentTokenizer` to tokenize the input string, extracting prefixes and their associated values. It ensures that all mandatory fields are present and that there are no duplicate prefixes in the user input.
 
 ##### Executing the Command
-A `AddCommand` is created by the `AddressBookParser` class and passed to the `Logic` component for execution. The `LogicManager` then calls the `execute` method in the `AddCommand` class.
+The `AddCommand` class creates a new `Person` object with the parsed details.
+The `Person` object is then added to the ``UniquePersonList` through the `addPerson` method in the `Model` component.
```
Comment (danielleloh) [inline docs/DeveloperGuide.md]: there is an extra ` at UniquePersonList

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -2331,34 +2363,101 @@ New fields such as recitation, studio, matriculation number, was added to person
...
+were also a lot more bug prone than others, such as the import features, which required extensive testing to ensure that
+all potential errors were caught and proper error messages were displayed. We also had to ensure that the application
+handled these errors gracefully and did not crash when these errors occurred.
 
 <br>
 
 ### Achievements
-Overall, our group successfully implemented the planned features while addressing bugs and managing potential feature flaws. 
+Overall, our group successfully implemented the planned features while addressing bugs and managing potential feature flaws.
 Despite initial hesitations about implementing significant new features like exams and exam scores, we overcame the challenge and achieved our goals.
```
Comment (Zer0Legion) [inline docs/DeveloperGuide.md:2463]: What an inspiring message.

**Review by Zer0Legion — COMMENTED** [review]
Much more consistent wording and formatting. Thanks Johan !

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -2331,34 +2363,101 @@ New fields such as recitation, studio, matriculation number, was added to person
...
+On the other hand, storing each person's score in their corresponding `Person` object would make it easier to implement
+operations on persons, but would require more complex interactions for exam management. We had to consider the pros and cons
+of each approach, before deciding on the latter approach, as we concluded that our application was more person-centric.
+
+#### Bug Fixing and Testing
+A significant challenge we faced was the identification and resolution of bugs. Unit tests for our own features were
+relatively straightforward to implement, but we found that identifying edge cases proved to be tricky. Certain features
+were also a lot more bug prone than others, such as the import features, which required extensive testing to ensure that
+all potential errors were caught and proper error messages were displayed. We also had to ensure that the application
+handled these errors gracefully and did not crash when these errors occurred.
```
Comment (danielleloh) [inline docs/DeveloperGuide.md:2457]: this writeup looks good and detailed to me :)
